packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -143,9 +145,9 @@ async function fetchLatestSnapshotCore(
     storageTokenFetcher: (options: TokenFetchOptions, name: string) => Promise<string | null>,
     snapshotOptions: ISnapshotOptions | undefined,
     logger: ITelemetryLogger,
-    snapshotDownloader: (url: string, fetchOptions: {[index: string]: any}) => Promise<IOdspResponse<IOdspSnapshot>>,
+    snapshotDownloader: <T>(url: string, fetchOptions: {[index: string]: any}) => Promise<IOdspResponse<T>>,
```

Comment:
I think we should remove template argument <T> here and use `unknown` return type - that much better reflects the intention here.

Reply:
But we know the type, then why unknown?

Reply:
The callback provided does not know the type. It's only the caller who knows the type

Reply:
Yes, that is why I provided template as the caller would supply the type of response it expects. Isn't it fine? Sorry I don't understand if you are saying something else.

Reply:
It's not wrong, but it is sort of cheating, as for function to be truely generic, it needs to depend on T somewhere (i.e. one of the args should depend on T). Here, it's a hidden cast, I'd rather see it more direct in this function.

Reply:
BTW, I started that cheating (in other places), but I think it's wrong :)